bdb-ip.md

```diff
@@ -0,0 +1,437 @@
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+##### Proposal 2
+
+- /transaction/ID is returning only transactions included in a VALID block
+
+
+Pros:
+
+- Expected behavior (similar to comparable systems: databases and blockchains
+alike)
+- Transaction's status can still be monitored at /status, as is expected
+- Non-persisted transaction cannot be queried from BDB
+- No mental overhead
+- Interface is safe: Users will be making mission-critical assumptions on the
+  /transactions/id endpoint but might miss the API docs part. We want it's
+  default behavior to be as safe as possible
+- Querying for UNDECIDED or BACKLOG is very difficult to achieve without
+  confusing the user (e.g. [transactions can be in multiple
+  states](https://github.com/bigchaindb/bigchaindb/issues/1038#issuecomment-274049790)).
+  Safes us and the user lots of time
+
+Contra:
+
+- Transactions from BACKLOG OR UNDECIDED blocks cannot be queried anymore
+(actually they can using /blocks and /votes)
+
+
+##### Proposal 3
+
+- /transaction/ID is returning only transactions included in a VALID block
+- /transaction/ID?safe=false (where ?safe=true or lack of is default) may
+return transactions from BAcKLOG or UNDECIDED blocks
+
+
+Pros:
+
+- Expected behavior (similar to comparable systems: databases and blockchains
+alike)
+- Non-persisted transaction can be queried from BDB (is that a Pro?)
+- Interface is safe: Users will be making mission-critical assumptions on the
+  /transactions/id endpoint but might miss the API docs part. We want it's
+  default behavior to be as safe as possible
+
+Contra:
+
+- Might be used as a substitute endpoint instead of /status?tx_id
+- Mental overhead
+- Ideally, the inner workings of BDB do not need to be fully understood in
+  order to work with BDB (compare to databases: Many database users aren't
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+#### Proposal 4
+
+- /transactions/id gets a ?status flag
+
+Contra:
+
+- Querying for UNDECIDED or BACKLOG is very difficult to achieve without
+  confusing the user (e.g. [transactions can be in multiple
+  states](https://github.com/bigchaindb/bigchaindb/issues/1038#issuecomment-274049790))
+  Safes us and the user lots of time
+- Might be used as a substitute endpoint instead of /status?tx_id
+- Ideally, the inner workings of BDB do not need to be fully understood in
+  order to work with BDB (compare to databases: Many database users aren't
+  experts on the details of the database's inner workings: "What is a BACKLOG,
+  and why should I care?")
+
+
+##### Personal favorite: Proposal 2
+
+
+#### [/transaction/id and /transaction?asset_id?operation=CREATE return same content](https://github.com/bigchaindb/bigchaindb/issues/1129)
+
+I can't think of any proposal. If anyone has a proposal that makes sense,
+please let me know.
```

Comment:
I think it would be okay to leave the longer form implemented so that there's no missing analogue to the TRANSFER version, but add a sentence or two to the documentation noting that they can get the same transaction (except not in a list) by using the `/transaction/id` version.